src/ol/source/ImageStatic.js

```diff
@@ -43,7 +43,48 @@ class Static extends ImageSource {
     const /** @type {import("../Image.js").LoadFunction} */ imageLoadFunction =
         options.imageLoadFunction !== undefined
           ? options.imageLoadFunction
-          : defaultImageLoadFunction;
+          : (image, src) => {
+              if (
+                crossOrigin !== null ||
+                location?.origin === new URL(src, location?.origin).origin
```

Comment:
`data` protocol should also be allowed here

Reply:
@mike-000 Can you elaborate? I see no reason why this logic wouldn't work with a `data:` URI.

Reply:
`new URL(src, location?.origin).origin` would return `null` in both Chrome and Firefox if `src` is a data url (unlike an object url which does return the location), and it does not need a `crossOrigin` setting so that can be `null`.

Reply:
Handling this in a different way, no need to `fetch()`.